uaclient/entitlements/fips.py

```diff
@@ -66,6 +66,30 @@ def _prevent_fips_on_xenial_cloud_instances(self):
 
         return bool(cloud_id in ("azure", "gce"))
 
+    def _prevent_default_fips_on_gcp_cloud(self) -> bool:
```

Comment:
It feels a bit like there is an opportinuty to parameterize this method and the other _prevent_fips_on_xenial_cloud_instances into a single method. We also have a bit of duplication in calls to get_platform_info and get_cloud_type that could be reduced if we pass those series and cloud_id values into this method.